src/main/java/client/Character.java

```diff
@@ -323,6 +325,45 @@ public void onAnnounceStatPoolUpdate() {
         setPosition(new Point(0, 0));
     }
 
+    static {
+        if (YamlConfig.config.server.USE_EXP_GAIN_LOG) {
+            SaveExpLogToDBThread expThread = new SaveExpLogToDBThread();
+            expThread.setPriority(Thread.MIN_PRIORITY); // Set to the lowest priority
+            expThread.start();
+        }
+    }
+
+    // Nested static class for saving exp logs to the database
+    private static class SaveExpLogToDBThread extends Thread {
+        @Override
+        public void run() {
+            while(true) {
+                try {
```

Comment:
This does not solve the issue with using a lot of database connections. Perhaps you could run a batch insert every X seconds?